org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/ContentModelCompletionParticipant.java

```diff
@@ -101,9 +112,26 @@ private void fillWithChildrenElementDeclaration(DOMElement element, Collection<C
 			CompletionItem item = new CompletionItem(label);
 			item.setFilterText(request.getFilterForStartTagName(label));
 			item.setKind(CompletionItemKind.Property);
-			String documentation = child.getDocumentation();
-			if (documentation != null) {
-				item.setDetail(documentation);
+			StringBuilder sb = new StringBuilder();

```

Comment:
I think this code should ne set on the cm document implementaion. I suggest you that you create an abstractcmdocument class which provides the getfulldocumentation method. This abstract class could store the uri too

Reply:
@angelozerr 

CMDTDDocument is already extending a class, while CMXSDDocument is not.

CMDTDElementDeclaration is also extending a class, while CMXSDElementDeclaration is not.

Is there something I could do to avoid this problem?

Reply:
Your right @NikolasKomonen please forget my comment

Reply:
A another suggestion is to set the documentation code with source in a static method in XMLGenerator. Set the build of documentation with source will help more how this doc is built.